reference/plugins.rst

```diff
@@ -0,0 +1,282 @@
+| | the context                       | ``export()`` | ``source()`` | ``build()`` | ``package()`` | | ``upload()``         | | ``download()``         |
+|                                     |              |              |             |               | | ``upload_recipe()``  | | ``download_recipe()``  |
+|                                     |              |              |             |               | | ``upload_package()`` | | ``download_package()`` |
++----------------+--------------------+--------------+--------------+-------------+---------------+------------------------+--------------------------+
+| **Parameters** | ``conanfile``      | Yes          | Yes          | Yes         | Yes           | No                     | post                     |
+|                +--------------------+--------------+--------------+-------------+---------------+------------------------+--------------------------+
+|                | ``conanfile_path`` | pre / post   | Yes          | user space  | pre / post    | Yes                    | post                     |
+|                +--------------------+--------------+--------------+-------------+---------------+------------------------+--------------------------+
+|                | ``reference``      | Yes          | cache        | cache       | cache         | Yes                    | Yes                      |
+|                +--------------------+--------------+--------------+-------------+---------------+------------------------+--------------------------+
+|                | ``package_id``     | No           | No           | cache       | Yes           | Yes                    | Yes                      |
+|                +--------------------+--------------+--------------+-------------+---------------+------------------------+--------------------------+
+|                | ``remote``         | No           | No           | No          | No            | Yes                    | Yes                      |
++----------------+--------------------+--------------+--------------+-------------+---------------+------------------------+--------------------------+
+
+\*Plugin functions are indicated without ``pre`` and ``post`` prefixes for simplicity.
+
+Table legend:
+  - **Yes**: Availability in ``pre`` and ``post`` functions in any context.
+  - **No**: Not available.
+  - **pre / post**: Availability in both ``pre`` and ``post`` functions with **different values**. e.g. ``conanfile_path`` pointing to user
+    space in ``pre`` and to local cache in ``post``.
+  - **post**: Only available in ``post`` function.
+  - **cache**: Only available when the context of the command executed is the local cache. e.g. :command:`conan create`,
+    :command:`conan install`...
+  - **user space**: Only available when the context of the command executed is the local cache. e.g. :command:`conan build`
+
```

Comment:
I wonder if this table is needed given the full example provided above.

Reply:
I consider it as a summary for clarification, I like to have the "usage" example above and the explanation in the table

Reply:
I was thinking about maintainability, we have to keep code and the documentation (in two places) synchronized. But it is ok, let's see if in the future this table grows too much.